consensus/consensus-types/src/common.rs

```diff
@@ -298,6 +353,21 @@ impl Payload {
                 }
                 Ok(())
             },
+            (true, Payload::QuorumStoreInlineHybrid(inline_batches, proof_with_status)) => {
+                for proof in proof_with_status.proof_with_data.proofs.iter() {
+                    proof.verify(validator)?;
+                }
+                for (batch, payload) in inline_batches.iter() {
+                    // TODO: Can cloning be avoided here?
+                    if BatchPayload::new(batch.author(), payload.clone()).hash() != *batch.digest()
```

Comment:
Possible to avoid cloning here?

Reply:
I'm getting an error if I don't clone though. Not sure if there is a better way.

Reply:
I see. We can leave it this way in that case.